File: front/app/js/services/TopicService.js
'use strict';
ngApp.factory(
  'TopicService',
  ['Restangular', 'AuthService',
    function (Restangular, AuthService) {
      return {
        getTopic: function(topicId, forEdit) {
          return Restangular.one('back/rest/topics', topicId).get({"forEdit": forEdit}).$object;
        },
        getSubjectTopics: function(subjectId) {
          if(subjectId != undefined) {
            return Restangular.one('back/rest/subjects/', subjectId).getList('topics').$object;
          } else {
            return [];
          }
        },
        createTopic: function(topic) {
          return Restangular.all('back/rest/topics').post(topic, {}, AuthService.getAuthHeaders());
        }
      };
    }
  ]
);

File: back/src/main/java/ee/teoreteetik/tt/service/TopicService.java
package ee.teoreteetik.tt.service;

import ee.teoreteetik.tt.model.Topic;
import java.util.List;

public interface TopicService {
  List<Topic> getBySubjectId(Long subjectId);

  Topic getById(Long topicId, boolean forEdit);

  Long createTopic(Topic topic);

}
